### DISCUSSION TITLE
Auto assignment of attrs if no load function

### DISCUSSION THREAD

**Discussion by gsxdsm** [tetra-framework/tetra#88] [Ideas]
If there is no load method, attributes specified on the component tag could be automatically assigned to public attributes on the python side. Thoughts?

**Comment by nerdoc** [comment]
Sure. This would even simplify coding more to have load() optional. I would avoid overriding the load method, as users would then need to call `super().load(...)` at every class.
Insteady, I introduced a `_pre_load()` method (to be overridden for such things) and played a bit with it, but it was not well thought, and I removed it again.

Maybe the `_call_load()` would do also, to put this logic in.
If you want to see a call graph of what happens in Tetra, I [created one in Figma](https://www.figma.com/board/s36Pc4HG52ASTgoE2PcKNB/Tetra-call-graph?node-id=0-1&node-type=canvas)

It should definitely be in a method that is called like load(), as these attrs should not be saved in states.

**Reply by gsxdsm** [reply]
Awesome. Adding this to my to-do list as well after form files.
